src/components/Button/PButton.vue

```diff
@@ -24,32 +27,31 @@
   import { RouteLocationRaw } from 'vue-router'
   import PIcon from '@/components/Icon/PIcon.vue'
   import PLoadingIcon from '@/components/LoadingIcon/PLoadingIcon.vue'
-  import { Icon } from '@/types/icon'
-  import { Size } from '@/types/size'
+  import { Icon, ButtonKind } from '@/types'
   import { isRouteExternal } from '@/utilities/router'
 
-  type ButtonClass = 'primary' | 'secondary' | 'inset' | 'flat' | 'danger' | 'danger-secondary'
 
   const props = defineProps({
-    secondary: Boolean,
-    inset: Boolean,
-    flat: Boolean,
-    danger: Boolean,
-    rounded: Boolean,
-    disabled: Boolean,
-    loading: Boolean,
-    size: {
-      type: String as PropType<Size>,
-      default: 'md',
+    kind: {
+      type: String as PropType<ButtonKind>,
+      default: 'default',
     },
     icon: {
       type: String as PropType<Icon>,
       default: undefined,
     },
+    iconAppend: {
+      type: String as PropType<Icon>,
+      default: undefined,
+    },
     to: {
       type: [String, Object] as PropType<RouteLocationRaw>,
       default: undefined,
     },
+    small: Boolean,
+    activated: Boolean,
+    disabled: Boolean,
+    loading: Boolean,
   })
```

Comment:
While we're updating components we should convert props over to how we do it in the other repos. 
```typescript
  const props = defineProps<{
    kind: ButtonKind,
    icon?: Icon,
    iconAppend?: Icon,
    to?: RouteLocationRaw,
    small?: boolean,
    activated?: boolean,
    disabled?: boolean,
    loading?: boolean,
  }>()
```